eventauth/eventcontent.go

```diff
@@ -0,0 +1,280 @@
+package eventauth
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
+// creaateContent is the JSON content of a m.room.create event along with
+// the top level keys needed for auth.
+type createContent struct {
+	// We need the domain of the create event when checking federatebility.
+	senderDomain string
+	// We need the roomID to check that events are in the same room as the create event.
+	roomID string
+	// The "m.federate" flag tells us whether the room can be federated to other servers.
+	Federate *bool `json:"m.federate"`
+	// The creator of the room tells us what the default power levels are.
```

Comment:
I wouldn't assume prior knowledge when it isn't self-explanatory - `creator` isn't clear if this is a user ID, server domain, or something completely different. A spec link would help clear up ambiguity and provide a reference for explaining what the field is for, so SGTM.